lms/services/blackboard_api.py

```diff
@@ -0,0 +1,60 @@
+from lms.validation.authentication import OAuthTokenResponseSchema
+
+
+class BlackboardAPIClient:
+    def __init__(
+        self,
+        blackboard_host,
+        client_id,
+        client_secret,
+        redirect_uri,
+        http_service,
+        oauth2_token_service,
+    ):  # pylint:disable=too-many-arguments
+        self.blackboard_host = blackboard_host
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.redirect_uri = redirect_uri
+
+        self._http_service = http_service
+        self._oauth2_token_service = oauth2_token_service
+
+    def get_token(self, authorization_code):
+        """
+        Get an access token from Blackboard and save it in the DB.
+
+        :raise services.HTTPError: if something goes wrong with the access
+            token request to Blackboard
+        """
```

Comment:
This function looks like it's covering a lot of the same ground as `services.canvas_api._authenticated.AuthenticatedClient.get_token()`. Are we re-inventing the wheel here?

I know that class isn't instantly ready for use, but it was intended to be adaptable. There isn't much specific to Canvas in there apart from one hard coded reference to `login/oauth2/token`.